aladdin.sh

```diff
@@ -85,22 +89,54 @@ function check_and_handle_init() {
 }
 
 function set_minikube_config(){
+    local vm_driver=$(jq --arg DEFAULT "$DEFAULT_MINIKUBE_VM_DRIVER" -r '.minikube.vm_driver//$DEFAULT' "$ALADDIN_CONFIG_FILE")
+    local memory=$(jq --arg DEFAULT "$DEFAULT_MINIKUBE_MEMORY" -r '.minikube.memory//$DEFAULT' "$ALADDIN_CONFIG_FILE")
+
     minikube config set kubernetes-version v$KUBERNETES_VERSION &> /dev/null
-    minikube config set vm-driver virtualbox &> /dev/null
-    minikube config set memory 4096 &> /dev/null
+    minikube config set vm-driver "$vm_driver" &> /dev/null
+    minikube config set memory "$memory" &> /dev/null
+}
+
+function _start_minikube() {
+    local minikube_cmd="minikube start"
+
+    if test "$OSTYPE" = "linux-gnu"; then
+        if test $(minikube config get vm-driver) = "none"; then
+            # If we're running on native docker on a linux host, minikube start must happen as root
+            # due to limitations in minikube.  Specifying CHANGE_MINIKUBE_NONE_USER causes minikube
+            # to switch users to $SUDO_USER (the user that called sudo) before writing out
+            # configuration.
+            minikube_cmd="sudo -E CHANGE_MINIKUBE_NONE_USER=true '$(which minikube)' start"
+
+        else
+            # On linux, /home gets mounted on /hosthome in the minikube vm, so as not to
+            # override /home/docker.  We still want the user's home directory to be in the
+            # same path both in and outside the minikube vm though, so symlink it into place.
+            minikube_cmd="$minikube_cmd &&\
+                          minikube ssh \"sudo mkdir '$HOME' && \
+                                         sudo mount --bind '${HOME/\/home//hosthome}' '$HOME'\""
+        fi
```

Comment:
This part looks fine to me, but maybe beneficial to get @bbil to look at this section?

Reply:
This is for virtual box right?

This looks like it is replacing the manual steps I used to do whenever creating a new minikube vm (through the virtual box UI).

Reply:
yup!